craft_parts/plugins/plugins.py

```diff
@@ -0,0 +1,107 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Definitions and helpers to handle plugins."""
+
+import copy
+from typing import TYPE_CHECKING, Any, Dict, Type
+
+from .autotools_plugin import AutotoolsPlugin
+from .base import Plugin
+from .dump_plugin import DumpPlugin
+from .make_plugin import MakePlugin
+from .nil_plugin import NilPlugin
+from .properties import PluginProperties
+
+if TYPE_CHECKING:
+    from craft_parts.infos import PartInfo
+    from craft_parts.parts import Part
+
+
+PluginType = Type[Plugin]
+
+
+# Plugin registry by plugin API version
+_BUILTIN_PLUGINS: Dict[str, PluginType] = {
+    "autotools": AutotoolsPlugin,
+    "dump": DumpPlugin,
+    "make": MakePlugin,
+    "nil": NilPlugin,
+}
+
+_PLUGINS = copy.deepcopy(_BUILTIN_PLUGINS)
```

Comment:
Perhaps consider (if you haven't already) an interface for the application to init craft-parts and receive a context to work with?  The plugin configuration would live in that context rather than a global var.   I imagine that would help for tests, etc.

Reply:
That's an interesting idea, I'll think about it. Currently only action execution runs in a context.